miio/yeelight.py

```diff
@@ -394,15 +513,39 @@ def set_name(self, name: str) -> bool:
         """Set an internal name for the bulb."""
         return self.send("set_name", [name])
 
-    @command(default_output=format_output("Toggling the bulb"))
-    def toggle(self):
+    @command(
+        click.option(
+            "--lamp", type=int, required=False, default=YeelightSubLightType.Main
+        ),
+        default_output=format_output("Toggling the bulb"),
+    )
+    def toggle(self, lamp=YeelightSubLightType.Main):
         """Toggle bulb state."""
-        return self.send("toggle")
+        if lamp == YeelightSubLightType.Background:
+            if not self.model_specs.has_background_light:
+                raise YeelightException("Background lamp not supported")
+            return self.send("bg_toggle")
+        elif (
+            lamp == YeelightSubLightType.Main
+            or not self.model_specs.has_background_light
+        ):
+            return self.send("toggle")
+        return self.send("dev_toggle")
 
-    @command(default_output=format_output("Setting current settings to default"))
-    def set_default(self):
+    @command(
+        click.option(
+            "--lamp", type=int, required=False, default=YeelightSubLightType.Main
+        ),
```

Comment:
Hello @rytilahti  sorry for a while, I was trying to figure out some good solution but failed.
Main problem that `--lamp <type>` in `get_device_group` pass to constructor but not a function property. And this is ruining using this functions not from command line. (Recreate a Yeelight device  or pass `target` property every time I want to change lamp status looks ugly).

As for me: better idea can be use some thing like sub device for every sub lights. 
I see this like main lamp device(general function like `dump_ble_debug` and general property like `save_state_on_change`) with one or two sub light device(Main and Background light). 
Can I know device model during constructor to create list of sub device? I need this not only for get status function. 
But get status for this 3(or 2) device must use one single request to physical lamp. 
If some one call `status` function on the sub light - system must check when was be last  request and use cashed data or send new one using general lamp device. 

But every my tried was failed. (because command-line part doesn't wont to work with sub light class. Only with main light.(through an error `Managed to invoke callback without a context`).

Could you provide me some work around about this.

PS I have a lot of work at my main place of work for at least the next three months, but I am not forgetting about this pull request.

Reply:
Hi @Kirmas and sorry for the delayed answer!

> As for me: better idea can be use some thing like sub device for every sub lights.

Agreed. A regular light and a background light should implement the same interface to allow easy interfacing.

> I see this like main lamp device(general function like `dump_ble_debug` and general property like `save_state_on_change`) with one or two sub light device(Main and Background light).

The main light (and its data) is always available so I think the sublight should re-use these.

> Can I know device model during constructor to create list of sub device?

 The model information is now available by just accessing `self.model`.

> If some one call status function on the sub light - system must check when was be last request and use cashed data or send new one using general lamp device.

I think that trying to access the status of the secondary light should just return the information from the main query which returns all necessary data already.